src/antsibull/collection_meta_lint.py

```diff
@@ -0,0 +1,207 @@
+            )
+
+        if removal.redirect_replacement_version is not None:
+            if removal.redirect_replacement_version <= self.major_release:
+                self.errors.append(
+                    f"{prefix} redirect_replacement_version: Redirect removal version"
+                    f" {removal.redirect_replacement_version} must be larger than"
+                    f" current major version {self.major_release}"
+                )
+            if (
+                removal.version != "TBD"
+                and removal.redirect_replacement_version >= removal.version
+            ):
+                self.errors.append(
+                    f"{prefix} redirect_replacement_version: Redirect removal version"
+                    f" {removal.redirect_replacement_version} must be smaller than"
+                    f" the removal major version {removal.version}"
+                )
+
+        if removal.reason == "renamed":
+            self._validate_reason_renamed(collection, removal, prefix)
+        else:
+            self._validate_reason_not_renamed(collection, removal, prefix)
+
+        if removal.reason == "other":
+            self._validate_reason_other(collection, removal, prefix)
+        else:
+            self._validate_reason_not_other(collection, removal, prefix)
+
+    def _validate_collection(
+        self, collection: str, meta: CollectionMetadata, prefix: str
+    ) -> None:
+        if meta.repository is None:
+            self.errors.append(f"{prefix} repository: Required field not provided")
+
+        if meta.removal:
+            self._validate_removal(collection, meta.removal, f"{prefix} removal ->")
+
+    def validate(self, data: CollectionsMetadata) -> None:
+        # Check order
+        sorted_list = sorted(data.collections)
+        raw_list = list(data.collections)
+        if raw_list != sorted_list:
+            for raw_entry, sorted_entry in zip(raw_list, sorted_list):
+                if raw_entry != sorted_entry:
+                    self.errors.append(
+                        "The collection list must be sorted; "
+                        f"{sorted_entry!r} must come before {raw_entry}"
+                    )
+                    break
+
+        # Validate collection data
+        remaining_collections = set(self.all_collections)
+        for collection, meta in data.collections.items():
+            if collection not in remaining_collections:
+                self.errors.append(
+                    f"collections -> {collection}: Collection not in ansible.in"
+                )
+            else:
+                remaining_collections.remove(collection)
+            self._validate_collection(
+                collection, meta, f"collections -> {collection} ->"
+            )
+
+        # Complain about remaining collections
+        for collection in sorted(remaining_collections):
+            print(f"collections: No metadata present for {collection}")
```

Comment:
Should this be an error?

Reply:
Ooops, yes...

Reply:
ed725859c9cc9035d594a1cb9452f1dafae79af1